File: backend/src/domain/entities/MentorshipSession.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('mentorship_sessions')
export class MentorshipSession {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'availability_id' })
  availabilityId: number;

  @Column({ name: 'mentee_member_id' })
  menteeMemberId: number;

  @Column({ type: 'varchar', nullable: true })
  status: string;

  @Column({ type: 'text', name: 'booking_note', nullable: true })
  bookingNote: string;

  @Column({ type: 'varchar', name: 'meeting_link', nullable: true })
  meetingLink: string;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/shared/entity/MentorshipSession.java
package com.service.backend.shared.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import com.service.backend.shared.enums.Status;
import com.service.backend.shared.enums.MentorshipSessionType;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("mentorship_sessions")
public class MentorshipSession {

    @Id
    private Integer id;

    @Column("availability_id")
    private Integer availabilityId;

    @Column("mentee_member_id")
    private Integer menteeMemberId;

    private Status status;

    @Column("booking_note")
    private String bookingNote;

    @Column("meeting_link")
    private String meetingLink;

    @Column("session_type")
    private MentorshipSessionType sessionType;

    private String introduction;

    private String description;

    @Column("cv_url")
    private String cvUrl;

    @Column("cancel_reason")
    private String cancelReason;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
